### PR TITLE
Use PHP_BINARY for server command execution

### PR DESCRIPTION
**Pull request by donatj**
Currently, we're dependent on PHP being in the $PATH of the user running the tests. 

This usually works, but something happened with macOS 26.3 and that seems to not always be the case in certain contexts such as running tests from PHPStorm and within other _Applications_ not running the tests from within a shell context.

We do not need to depend on PATH for the php executable and can instead get the binary running the tests to begin with, `PHP_BINARY`

This also prevents a weird edge case where the PHP version running the test might not be the same version running the server.

----

> [!IMPORTANT]
> Use `PHP_BINARY` in `MockWebServer.php` to ensure consistent PHP version for server execution.
> 
>   - **Behavior**:
>     - Modify `start()` in `MockWebServer.php` to use `PHP_BINARY` for server command execution instead of `php` in the system's PATH.
>     - Ensures the PHP version running the tests matches the server's PHP version, preventing version mismatch issues.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=donatj%2Fmock-webserver&utm_source=github&utm_medium=referral)<sup> for 9ac348c8ee8fbb247a4131d0dc7d3802f3e86462. You can [customize](https://app.ellipsis.dev/donatj/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: src/MockWebServer.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ public function start() : void {
 		$script = __DIR__ . DIRECTORY_SEPARATOR . '..' . DIRECTORY_SEPARATOR . 'server' . DIRECTORY_SEPARATOR . 'server.php';
 
 		$stdout = tempnam(sys_get_temp_dir(), 'mockserv-stdout-');
-		$cmd    = sprintf("php -S %s:%d %s", $this->host, $this->port, escapeshellarg($script));
+		$cmd    = sprintf("%s -S %s:%d %s", PHP_BINARY, $this->host, $this->port, escapeshellarg($script));
```
[the remark this answers is not in the document]
Comment (donatj) [reply src/MockWebServer.php]: Good catch.
